docs/android/linking.md

```diff
@@ -0,0 +1,155 @@
+
+<!-- markdownlint-disable MD025 -->
+# [Visual Studio](#tab/vs)
+<!-- markdownlint-enable MD025 -->
+
+1. In **Solution Explorer** right-click on your .NET MAUI app project and select **Properties**. Then, navigate to the **Android > Options** tab and ensure that trimming is enabled for the release build configuration:
+
+    :::image type="content" source="media/linking/vs.png" alt-text="Screenshot of the linker behavior for Android in Visual Studio.":::
+
+<!-- markdownlint-disable MD025 -->
+# [Visual Studio for Mac](#tab/vsmac)
+<!-- markdownlint-enable MD025 -->
+
+1. In the **Solution Window**, right-click on your .NET MAUI app project and select **Properties**.
+1. In the **Project Properties** window, select the **Build > Android > Linker** tab.
+1. In the **Project Properties** window, ensure the **Configuration** drop-down is set to **Release** and set the **Linker Behavior** drop-down to your desired linker behavior:
+
+    - *Don't link*. Disabling linking ensures assemblies aren't modified.
+    - *Link SDK assemblies only*. In this mode, the linker leaves your assemblies untouched and reduces the size of the SDK assemblies by removing types and members that your app doesn't use.
+    - *Link all assemblies*. When it links all assemblies, the linker performs additional optimizations to make your app as small as possible. It modifies the intermediate code for your source code, which may break your app if you use features using an approach that the linker's static analysis can't detect. In these cases, you may need to make adjustments to your source code to make your app work correctly.
+
+    :::image type="content" source="media/linking/vsmac.png" alt-text="Screenshot of the linker behavior for Android in Visual Studio for Mac.":::
+
+    > [!WARNING]
+    > Linking all assemblies isn't recommended, and can require significant additional work to ensure that your app still works.
+
+1. In the **Project Properties** window, click the **OK** button.
+
+> [!NOTE]
+> Visual Studio for Mac sets the `$(AndroidLinkMode)` MSBuild property in your app's project file. `<AndroidLinkMode>None</AndroidLinkMode>` maps to `<PublishTrimmed>false</PublishTrimmed>` and `<AndroidLinkMode>SdkOnly</AndroidLinkMode>` maps to `<PublishTrimmed>true</PublishTrimmed>`.
+
+----
+
+## Preserve code
+
+When you use the linker, it sometimes remove code that you might have called dynamically, even indirectly. You can instruct the linker to preserve members by annotating them with the [`DynamicDependency`](xref:System.Diagnostics.CodeAnalysis.DynamicDependencyAttribute) attribute. This attribute can be used to express a dependency on either a type and subset of members, or at specific members.
+
+> [!IMPORTANT]
+> Every member that isn't statically linked by the app is subject to be removed.
```

Comment:
Using the phrase "statically linked" sounds like a C/C++ term to me.

Maybe we could phrase it as:

```suggestion
> Every member in the BCL that cannot be statically determined to be used by the app is subject to be removed.
```
That still sounds wordy to me... But maybe something like that?